### PR TITLE
refactor_:  move partial config relate to local pair sync to backend 

### PR DESCRIPTION
**Pull request by qfrank**
relate status-go [PR](https://github.com/status-im/status-go/pull/5080) 

### Testing notes
local pair sync should work for following direction:
- ios->android 
- ios->ios

#### Platforms

- Android
- iOS

status: ready

### CODE REVIEW

**Review by siddarthkay — APPROVED** [review]

**Comment by pavloburykh** [comment]
@qfrank please, ping here once PR is rebased and ready for testing. Thank you.

**Comment by pavloburykh** [comment]
Hey @churik! Could you please take a look at this PR. It will require testing sync between two IOS devices as well. I do not have possibility to test between 2 ios devices so unassigning myself. Thank you.

File: src/legacy/status_im/node/core.cljs [hunk 1]
```diff
@@ -117,10 +116,7 @@
         {:keys [installation-id log-level
                 waku-bloom-filter-mode]}
         profile]
-    (cond-> {:NetworkId      (chain/chain-keyword->chain-id :mainnet)
-             :DataDir        "/ethereum/mainnet_rpc"
-             :UpstreamConfig {:Enabled true
-                              :URL     config/mainnet-rpc-url}}
+    (cond-> {}
```
Comment (ibrkhalil) [inline src/legacy/status_im/node/core.cljs:119]: Why do we need `cond->` with an `:always` :key on an empty map?

**Review by Samyoul — APPROVED** [review]

File: modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt [hunk 2]
```diff
@@ -47,7 +47,8 @@ class AccountManager(private val reactContext: ReactApplicationContext) : ReactC
 
     private fun updateConfig(jsonConfigString: String, absRootDirPath: String, keystoreDirPath: String): String {
         val jsonConfig = JSONObject(jsonConfigString)
-        val dataDirPath = jsonConfig.getString("DataDir")
+        // when doing local pair syncing, backend will provide default data dir
+        val dataDirPath = jsonConfig.optString("DataDir","")
```
Comment (ibrkhalil) [inline modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt:51]: Is it possible that there's a better default path than an empty string?

**Review by ibrkhalil — APPROVED** [review]

File: modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt (hunk 2 above)
Comment (qfrank) [reply modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt:51]: empty value is only for local pair sync, backend will provide default value to override empty value. @ibrkhalil

File: src/legacy/status_im/node/core.cljs (hunk 1 above)
Comment (qfrank) [reply src/legacy/status_im/node/core.cljs:119]: if you have better code structure, do let me know, thank you @ibrkhalil

**Comment by qfrank** [comment]
ready for manual QA 🙂 @qoqobolo

**Comment by qoqobolo** [comment]
> ready for manual QA 🙂 @qoqobolo

Thanks @qfrank! I'll check it as soon as I get a chance 🙏

File: modules/react-native-status/ios/RCTStatus/AccountManager.m [hunk 3]
```diff
@@ -66,6 +66,9 @@ -(NSString *) prepareDirAndUpdateConfig:(NSString *)config
     NSData *configData = [config dataUsingEncoding:NSUTF8StringEncoding];
     NSDictionary *configJSON = [NSJSONSerialization JSONObjectWithData:configData options:NSJSONReadingMutableContainers error:nil];
     NSString *relativeDataDir = [configJSON objectForKey:@"DataDir"];
+    if (relativeDataDir == nil) {
+        relativeDataDir = @"";
+    }
```
Comment (siddarthkay) [inline modules/react-native-status/ios/RCTStatus/AccountManager.m:71]: nice

**Review by siddarthkay — APPROVED** [review]

**Comment by qoqobolo** [comment]
Thanks for the PR @qfrank!

I checked sync between:
iOS <-> iOS
Android <-> iOS
Android <-> Desktop
iOS <-> Desktop

No issues found, PR can be merged.

**Pull request merged**